imod/mf6/buy.py

```diff
@@ -89,7 +90,7 @@ class Buoyancy(Package):
 
     _pkg_id = "buy"
     _template = Package._initialize_template(_pkg_id)
-    _metadata_dict = {}
+    _metadata_dict: Dict[str, VariableMetaData] = {}
```

Comment:
I see this ``_metadata_dict`` attribute is implemented for MetaSWAP, but not at all applied for Modflow 6. I see it is possibly used in the private method ``_get_vars_to_check``, but this one is never called, so I think we can remove it! I guess we once removed it, but this one was leftover?